### ISSUE TITLE
[Question] How to react to parent AlphaFilter changes in a deeply nested child?

### ISSUE THREAD

**Issue by IljaDaderko** [pixijs/pixi.js#6966]
I have a custom class that extends `PIXI.Graphics`. This class is responsible for creating dom elements like `<input />` for example and modifying such dom elements style when associated `PIXI.Graphics` instance changes.

Thus far I have been extending `updateElement` method and it worked great for alpha / scale / x / y transforms. In this case `this.element` is a reference to DOM `<input />` element.

```ts
private updateElement() {
    this.element.style.opacity = `${this.worldAlpha}`;
    const { a, b, c, d, tx, ty } = this.worldTransform;
    this.element.style.left = '0px';
    this.element.style.right = '0px';
    this.element.style.width = `${this.width}px`;
    this.element.style.height = `${this.height}px`;
    this.element.style.transform = `matrix(${a}, ${b}, ${c}, ${d}, ${tx}, ${ty})`;
  }
```

I recently came across same issue with `alpha` property as #4334 where changing alpha of container also changes alpha of its children. So I changed my implementation to use `PIXI.filters.AlphaFilter` and unfortunately this broke `updateElement` function above, because `worldAlpha` is now always the same: 1.

I am trying to figure out how to modify that function to instead rely on AlphaFilter value now. Issue is that children can be nested deeply within container that is actually using AlphaFilter, so I'm not sure how to get correct value and does it even trigger  `updateElement` ?

**Comment by bigtimebuddy** [comment]
There's unfortunately not a convenient, built-in way to get a parents AlphaFilter `alpha` like `worldAlpha`. But there are a bunch of ways you could workaround this, either by passing the alpha value to `updateElement` whenever it updates, creating your own sub-classes with something that behaves like worldAlpha (e.g. `worldFilterAlpha`) or to traverse up the tree until you get an AlphaFilter:

```js
let parent = this.parent;
const findFilter = f => f instanceof AlphaFilter;
while(parent) {
  const alphFilter = parent.filters.find(findFilter);
  if (alphFilter) {
    this.element.style.opacity = alphFilter.alpha;
    break;
  }
  parent = parent.parent;
}
```

Kinda all depends on your setup.

Hope that helps.

**Comment by IljaDaderko** [comment]
@bigtimebuddy yeh, this does the trick. I ran across one issue where child could have 2 parents with alpha filter, i.e. PageContainer and LayoutContainer, in case where LayoutContainer was animating, it had no effect as while loop hit `break` condition prior to that and stayed at PageContainers value.

I was able to solve this by modifying name of the instance that was animating and adding check for that in the loop. idk if it's the best solution, but it did the trick.
